### PR TITLE
VZ-6112 Jaeger instance should be creatable during VZ Install

### PR DESCRIPTION
**Pull request by ayaranga**
Changes are done to create a default Jaeger instance inside the mesh when Jaeger Operator is enabled via Verrazzano CR. The default Jaeger instance uses OpenSearch, installed as part of Verrazzano, for storage.

Fixes VZ-6112.

### CODE REVIEW

File: platform-operator/controllers/verrazzano/component/common/overrides.go [hunk 1]
```diff
@@ -5,6 +5,7 @@ package common
 
 import (
 	"context"
+	"github.com/Jeffail/gabs/v2"
```
[the remark this answers is not in the document]
Comment (ayaranga) [reply platform-operator/controllers/verrazzano/component/common/overrides.go:8]: It's already being used in many files. Example: platform-operator/controllers/clusters/sync_prometheus.go

[the remark this answers is not in the document]
Comment (kminder) [reply platform-operator/controllers/verrazzano/component/common/overrides.go:8]: I confirmed this is ok

**Review by kminder — APPROVED** [review]
Approved.

File: platform-operator/controllers/verrazzano/component/jaeger/operator/jaeger_operator.go [hunk 2]
```diff
@@ -201,3 +267,83 @@ func generateOverridesFile(ctx spi.ComponentContext, contents []byte) (string, e
 		string(contents))
 	return overridesFileName, nil
 }
+
+// createJaegerSecret creates a Jaeger secret for Elastic storage credentials.
```
[the remark this answers is not in the document]
Comment (ayaranga) [reply platform-operator/controllers/verrazzano/component/jaeger/operator/jaeger_operator.go:271]: Ok, will create a separate PR to fix this. Thanks.

File: platform-operator/controllers/verrazzano/component/jaeger/operator/jaeger_operator.go [hunk 3]
```diff
@@ -52,6 +62,27 @@ const extraEnvValueTemplate = `extraEnv:
...
 
+// A template to define Jaeger override
+// As Jaeger Operator helm-chart does not use tpl in rendering Jaeger spec value, we can not use
+// jaeger-operator-values.yaml override file to define Jaeger value referencing other values.
+const jaegerValueTemplate = `jaeger:
+  create: true
+  spec:
+    annotations:
+      sidecar.istio.io/inject: "true"
+    strategy: production
```
Comment (pfmackin) [inline platform-operator/controllers/verrazzano/component/jaeger/operator/jaeger_operator.go:73]: can we override production, or does Verrazzano Jaeger always require OpenSearch?

File: platform-operator/helm_config/charts/verrazzano/templates/networkpolicy.yaml [hunk 4]
```diff
@@ -37,6 +37,16 @@ spec:
       ports:
         - protocol: TCP
           port: 8775
+    - from:
```
Comment (pfmackin) [inline platform-operator/helm_config/charts/verrazzano/templates/networkpolicy.yaml:40]: why is Jaeger talking to the authproxy?

File: platform-operator/controllers/verrazzano/component/jaeger/operator/jaeger_operator.go (hunk 3 above)
Comment (ayaranga) [reply platform-operator/controllers/verrazzano/component/jaeger/operator/jaeger_operator.go:73]: The entire spec can be overriden as part of helm Overrides in VZ CR. VZ-6333 tracks supporting Jaeger all-in-one image. This is required for supporting inmemory strategy. The other strategy that is supported by Jaeger is streaming. That should work similar to production.

File: platform-operator/helm_config/charts/verrazzano/templates/networkpolicy.yaml (hunk 4 above)
Comment (ayaranga) [reply platform-operator/helm_config/charts/verrazzano/templates/networkpolicy.yaml:40]: This is required to access the url http://verrazzano-authproxy-elasticsearch.verrazzano-system.svc.cluster.local:8775.  Is this change not required? I was just following fluentd changes. Need to test if authorizationpolicy changes alone is sufficient.

**Pull request merged**